packages/cosmic-swingset/test/make.test.js

```diff
@@ -218,9 +218,100 @@ const walletProvisioning = test.macro({
       retryCount < retryCountMax,
       `wallet is provisioned within ${retryCount} retries out of ${retryCountMax}`,
     );
-
     await verifier?.stop(t);
   },
 });
 
 test.serial(walletProvisioning, 'wallet provisioning');
+
+/**
+ * getMetrics reads data from the Prometheus URL and returns the selected
+ * metrics as an object.
+ *
+ * @param {string} url            Prometheus URL
+ * @param {string[]} metricNames  Selected metric names
+ * @returns {Promise<Record<string, number>>}
+ */
+const getMetrics = async (url, metricNames) => {
+  const metricsResponse = await fetch(url);
+  if (!metricsResponse.ok) {
+    throw new Error(`Failed to fetch metrics: ${metricsResponse.statusText}`);
+  }
+
+  const metricsText = await metricsResponse.text();
+
+  // https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format
+  // metric_name [
+  //   "{" label_name "=" `"` label_value `"` { "," label_name "=" `"` label_value `"` } [ "," ] "}"
+  // ] value [ timestamp ]
+  const metricNamePatt = '[a-zA-Z_:][a-zA-Z0-9_:]*';
+  const labelNamePatt = '[a-zA-Z_][a-zA-Z0-9_]*';
+  const labelPatt = String.raw`${labelNamePatt}="(?:[^\\"\n]|\\(?:\\|"|n))*"`;
+  const samplePatt = RegExp(
+    String.raw`^(${metricNamePatt}(?:[{]${labelPatt}(?:,${labelPatt})*,?[}])?) +(\S+)( +-?[0-9]+)?$`,
+    'u',
+  );
+  /** @type {Iterable<any, any>} */
+  const allMetricsEntries = metricsText
+    .split('\n')
+    .map(line => {
+      const [_, metricName, value] = line.match(samplePatt) || [];
+      if (!metricName) return;
+      if (value === 'NaN') return [metricName, NaN];
+      if (value === '+Inf') return [metricName, Infinity];
+      if (value === '-Inf') return [metricName, -Infinity];
+      const valueNum = parseFloat(value);
+      if (Number.isNaN(valueNum)) {
+        throw Error(`${value} is not a decimal value`);
+      }
+      return [metricName, valueNum];
+    })
+    .filter(entry => !!entry);
```

Comment:
> I'll keep the `@type {Iterable<any, any>}` for now.

Actually, please don't.
```suggestion
  const allMetricsEntries = /** @type {Array<[string, number]>} */ (
    metricsText
      .split("\n")
      .map((line) => {
        const [_, metricName, value] = line.match(samplePatt) || [];
        if (!metricName) return undefined;
        if (value === "NaN") return [metricName, NaN];
        if (value === "+Inf") return [metricName, Infinity];
        if (value === "-Inf") return [metricName, -Infinity];
        const valueNum = parseFloat(value);
        if (Number.isNaN(valueNum)) {
          throw Error(`${value} is not a decimal value`);
        }
        return [metricName, valueNum];
      })
      .filter((entry) => !!entry)
  );
```